src/Stratis.Bitcoin.Features.Api/ApiFeature.cs

```diff
@@ -58,7 +58,21 @@ public override void Initialize()
             this.logger.LogInformation("API starting on URL '{0}'.", this.apiSettings.ApiUri);
             this.webHost = Program.Initialize(this.fullNodeBuilder.Services, this.fullNode, this.apiSettings);
 
-            this.TryStartKeepaliveMonitor();
+            // Start the keepalive timer, if set.
+            // If the timer expires, the node will shut down.
+            if (this.apiFeatureOptions.KeepaliveTimer != null)
+            {
+                this.apiFeatureOptions.KeepaliveTimer.Elapsed += (sender, args) =>
+                {
+                    this.logger.LogInformation($"The application will shut down because the keepalive timer has elapsed.");
+
+                    this.apiFeatureOptions.KeepaliveTimer.Stop();
```

Comment:
I suggest adding this lines in the feature Dispose method as well.
```
  if (this.apiFeatureOptions.KeepaliveTimer != null)
  {
                   this.apiFeatureOptions.KeepaliveTimer.Stop();
                   this.apiFeatureOptions.KeepaliveTimer.Enabled = false;
```

Also can we call a dispose on the timer? (is the timer disposable?)

Reply:
good idea :-)